### PR TITLE
Optimised Copy point Cloud

### PR DESCRIPTION
**Pull request by MMiknis**
CopyPointCloud method makes a few redundant calls for resizing even when it is not needed.
Using pointers and pointer incrementation allows for faster access when mem copying the cloud.

Under VS2013 compiler with full optimization enabled got these results.
Stock: CopyPointCloud -  17.8% cpu usage - 58fps
Opti: CopyPointCloud - 14.1% cpu usage - 68fps

### CODE REVIEW

**Comment by VictorLamoine** [comment]
Please use the PCL style guide. You can find it [here](http://pointclouds.org/documentation/advanced/pcl_style_guide.php).
There is an [Eclipse formatter file](https://github.com/PointCloudLibrary/pcl/blob/master/doc/advanced/content/files/PCL_eclipse_profile.xml) if you want to auto-format your code.

The best would be that you do a `git rebase -i upstream/master` and edit your two commits, then format correctly the code, use `commit --amend` and push again. (tell me if you need more details!)

I'll write a few comments in your code showing what's wrong.

File: common/include/pcl/common/impl/io.hpp [hunk 1]
```diff
@@ -118,18 +118,27 @@ pcl::copyPointCloud (const pcl::PointCloud<PointInT> &cloud_in,
   cloud_out.is_dense = cloud_in.is_dense;
   cloud_out.sensor_orientation_ = cloud_in.sensor_orientation_;
   cloud_out.sensor_origin_ = cloud_in.sensor_origin_;
-  cloud_out.points.resize (cloud_in.points.size ());
+  
 
-  if (cloud_in.points.size () == 0)
-    return;
+  const int inPointSize = cloud_in.points.size();
```
Comment (VictorLamoine) [inline common/include/pcl/common/impl/io.hpp]: Missing space before parenthesis: `size()` -> `size ()`

**Comment by MMiknis** [comment]
Sorry once again, I will learn from my mistakes one day.

File: common/include/pcl/common/impl/io.hpp [hunk 2]
```diff
@@ -118,18 +118,27 @@ pcl::copyPointCloud (const pcl::PointCloud<PointInT> &cloud_in,
...
-      copyPoint (cloud_in.points[i], cloud_out.points[i]);
+  { 
+    auto inPoints = &cloud_in.points[0];   //Create pointer for incoming points
+    auto outPoints = &cloud_out.points[0]; //Create pointer for outgoing points
+    int memSize = sizeof(PointOutT);
+    if (sizeof(PointInT) < memSize)
+      memSize = sizeof(PointInT);
+	// Iterate over each point
+	for (size_t i = 0; i < inPointSize; ++i)
+	  memcpy(outPoints++, inPoints++, memSize);
```
Comment (VictorLamoine) [inline common/include/pcl/common/impl/io.hpp]: Wrong indentation or worse: missing brackets :wink:

**Comment by taketwo** [comment]
This "optimization" is plain wrong. The `copyPoint()` function is used for a reason. It takes into account what fields are present in each point type (and in which order!) and transfers bytes accordingly. And it is smart. When appropriate, it uses `memcpy`. I encourage you to check the implementation.

Besides, please pay attention to the Travis results. Your proposed code does not compile (because PCL does not use C++11). If it compiled though, we would probably see a relevant unit test failing because of the changes.

**Comment by MMiknis** [comment]
Thanks for the feedback guys, I will go back to the drawing board and fix things :)

**Pull request closed without merging**